Jint/Native/Number/Dtoa/CachePowers.cs

```diff
@@ -52,17 +50,15 @@ internal CachedPower(ulong significand, short binaryExponent, short decimalExpon
         }
 
 
-        internal static int GetCachedPower(int e, int alpha, int gamma, DiyFp cMk)
+        internal static (short, DiyFp) GetCachedPower(int e, int alpha, int gamma)
         {
             const int kQ = DiyFp.KSignificandSize;
             double k = System.Math.Ceiling((alpha - e + kQ - 1)*Kd1Log210);
             int index = (GrisuCacheOffset + (int) k - 1)/CachedPowersSpacing + 1;
             CachedPower cachedPower = CACHED_POWERS[index];
 
-            cMk.F = cachedPower.Significand;
-            cMk.E = cachedPower.BinaryExponent;
-            Debug.Assert((alpha <= cMk.E + e) && (cMk.E + e <= gamma));
-            return cachedPower.DecimalExponent;
+            //System.Diagnostics.Debug.Assert((alpha <= cMk.E + e) && (cMk.E + e <= gamma));
```

Comment:
Did you intend to remove the assert?

Reply:
Mostly for being a bit lazy. It used calculations using mutable state that no longer exists with DiyFp being a struct, I think I can return them with some tweaks.